### PR TITLE
chore(styles, docs): re-apply mask icon changes

### PR DESCRIPTION
**Pull request by alizedebray**
Used mask-image icons for all components except form checks that will be handled separately.

### CODE REVIEW

File: packages/styles/src/components/_index.scss [hunk 1]
```diff
@@ -45,4 +45,4 @@
 @use 'datepicker';
 @use 'dropdown';
 @use 'floating-label';
-@use 'form-feedback';
+@use 'form-validation';
```
Comment (oliverschuerch) [inline packages/styles/src/components/_index.scss:48]: Is this part of this PR or should we move it to an own PR?

File: packages/styles/src/components/datatable.scss [hunk 2]
```diff
@@ -67,22 +68,18 @@
...
-    background-position: center top, center bottom;
+    @include icons-mx.icon(2127);
   }
 
-  &.datatable-icon-up {
-    background-image: datatable.$datatable-sort-asc-icon;
-    background-position: center center;
+  &.datatable-icon-up,
+  &.datatable-icon-down {
+    @include icons-mx.icon(2112);
```
Comment (oliverschuerch) [inline packages/styles/src/components/datatable.scss:78]: What do you think, should we document this mixin on the icons docs page?

File: packages/styles/src/components/datepicker.scss [hunk 3]
```diff
@@ -133,7 +136,8 @@ span.ngb-dp-navigation-chevron {
   background: transparent;
 
   .pi-3203,
-  .pi-calendar {
+  .pi-calendar,
+  post-icon {
     @include icons-mx.pi(3203);
```
Comment (oliverschuerch) [inline packages/styles/src/components/datepicker.scss:141]: Can we change this to `icons-mx.icon(3203)`?

File: packages/styles/src/mixins/_animation.scss [hunk 4]
```diff
@@ -31,6 +31,7 @@
   }
 }
 
+// DEPRECATED
```
Comment (oliverschuerch) [inline packages/styles/src/mixins/_animation.scss:34]: Did you add this here: https://github.com/swisspost/design-system/issues/1215?

File: packages/styles/src/variables/components/_datatable.scss [hunk 5]
```diff
@@ -45,9 +38,12 @@ $datatable-button-width: calc(100% - 2 * #{$datatable-button-margin});
...
 $datatable-button-focus-shadow: button.$btn-focus-box-shadow;
 
-$_pencil: icons.get-colored-svg-url('3193', color.$gray-80);
-$datatable-editable-icon: url($_pencil);
-
 $datatable-progressbar-height: spacing.$size-micro;
 $datatable-progressbar-backgroundcolor: color.$gray-10;
 $datatable-progressbar-activecolor: color.$yellow;
+
+// DEPRECATED
```
Comment (oliverschuerch) [inline packages/styles/src/variables/components/_datatable.scss:45]: Did you add this here: https://github.com/swisspost/design-system/issues/1215?

**Review by oliverschuerch — DISMISSED** [review]
There are lot of changes, which do not directly have a relation to the re-apply mask icon changes in my opinion.
For example the form-feedback/form-validation renaming.
It would be nice if we could split this PR up in smaller ones, so it's easier to review and we can add suitable changesets.

File: packages/documentation/src/stories/components/card/card.stories.ts
[the remark this answers is not in the document]
Comment (alizedebray) [reply packages/documentation/src/stories/components/card/card.stories.ts:1]: I don't think it's necessary here since this story is just to show that you can use any content from a card.
However, I added this hint to other components having the "Icon" control (buttons and toats).

File: packages/styles/src/components/datatable.scss (hunk 2 above)
Comment (alizedebray) [reply packages/styles/src/components/datatable.scss:78]: I don't think this mixin should be used externally.
We use it internally because styles need to display icons without being able to modify the HTML structure, but this is not the case for Design System users who need to use the post-icon component.

File: packages/styles/src/components/_index.scss (hunk 1 above)
Comment (alizedebray) [reply packages/styles/src/components/_index.scss:48]: This PR is to merge back changes from this branch: https://github.com/swisspost/design-system/compare/main...re-applied-icon-changes. It includes the renaming of "form-feedback" to "form-validation".

**Comment by alizedebray** [comment]
> There are lot of changes, which do not directly have a relation to the re-apply mask icon changes in my opinion. For example the form-feedback/form-validation renaming. It would be nice if we could split this PR up in smaller ones, so it's easier to review and we can add suitable changesets.

@oliverschuerch The goal of this PR is to merge back changes that were reverted a few weeks back and are now contained on this branch https://github.com/swisspost/design-system/compare/main...re-applied-icon-changes. There were several PRs originally which explains why the changes are disparate. 

The rollback was only due to issues found on the form check styles, so we decided during a Roundtable to address them in a separate ticket (#1953) and re-apply all other changes in a single PR (this one).

**Review by imagoiq — APPROVED** [review]
Maybe just fix the 3 code smells. e2e seem to failing in other PRs as well I'm having a look
